rust/tools/authorization-logic/transparent_release_prototype/transparent_release.auth_logic

```diff
@@ -0,0 +1,276 @@
+      :- isSymbol(some_object).
+
+    "Binary_OakFunctionsLoader" canActAs "OakFunctionsLoader" :-
+        "Binary_OakFunctionsLoader"
+            hasProvenance("ProvenanceFile_OakFunctionsLoader"),
+        "EndorsementFile_OakFunctionsLoader" canActAs "ValidRekorEntry",
+        expected_hash("Binary_OakFunctionsLoader", binary_hash),
+        measured_hash("Binary_OakFunctionsLoader", binary_hash),
+        expected_hash("ProvenanceFile_OakFunctionsLoader", provenance_hash),
+        measured_hash("ProvenanceFile_OakFunctionsLoader", provenance_hash).
+
+    // This we hopefully can get rid of for the real version:
+    isSymbol("6de656f11599c11548519af6b7385504ed845a4ff6600745cb3d9e174b8956af").
+    isSymbol("cb255dbbea86b128379995ff7dc805fd026e2d21c997a8df688ba9e5cb98e1c1").
+    isSymbol("ProvenanceFile_OakFunctionsLoader").
+    isSymbol("Binary_OakFunctionsLoader").
+    isSymbol("EndorsementFile_OakFunctionsLoader").
+}
+
+//-----------------------------------------------------------------------------
+// Endorsement File Wrapper
+//-----------------------------------------------------------------------------
+
+// This is an output from the wrapper for endorsement files. This 
+// wrapper is a standalone binary with a CLI that takes these arguments:
+// - a path to an endorsement file
+// - a name for the application
+// and it outputs authorization that looks like the code below
+
+// To work in comparisons in the generated authorization logic, the times need 
+// to be integers that increase as the time advances. A way to do this is 
+// to represent the time as "Unix Epoch Seconds". The wrapper for the 
+// endorsement file needs to convert the times it parses into this format. This 
+// URL gives code for doing this in many languages: https://www.epochconverter.com/
+// under the header "Convert from human-readable date to epoch".
+
+// For this output these arguments are:
+// - endorsement file --> "endorsement_file.toml"
+// - application name --> "OakFunctionsLoader"
+// And the expiry time and release time are converted into epoch
+//
+//
+// # this is an endorsement file for the Oak Functions Loader
+// [[endorsement]]
+// binary_hash = "cb255dbbea86b128379995ff7dc805fd026e2d21c997a8df688ba9e5cb98e1c1"
+// provenance_file = "https://github.com/project-oak/oak/blob/provenances/cb255dbbea86b128379995ff7dc805fd026e2d21c997a8df688ba9e5cb98e1c1.toml"
+// provenance_file_hash = "58c705880605d2f0eda58f9403811277ea2a31128223532ad93e778cb330af0f"
+// release_timestamp = "2021-10-21T19:14:57"
+// expiry_timestamp = "2021-11-21T19:14:57"
+
+// Endorsement File Wrapper Output
+"EndorsementFile_OakFunctionsLoader" says {
+    expected_hash("ProvenanceFile_OakFunctionsLoader",
+        "6de656f11599c11548519af6b7385504ed845a4ff6600745cb3d9e174b8956af")
+           :- RealTimeIs(real_time),        
+              greater_than(real_time, "1634843697"),  // Release time
+              less_than(real_time, "1637522097").     // expiry time
```

Comment:
If we go with my suggestion above, I think this would become 

```
"6de656f11599c11548519af6b7385504ed845a4ff6600745cb3d9e174b8956af" canActAs "ProvenanceFile_OakFunctionsLoader" :- ...
```